backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -119,7 +119,7 @@ public void initCounters() {
                     AtomicInteger::new,
                     state -> fetcher.apply(state.getAndIncrement())
                 )
-                .until(page -> page.getData().isEmpty())
+                .whilst(page -> page.getData().size() == rbacElementsPerPage)
```

Comment:
Most of the time, this will prevent one RBAC request per recipients resolution.

See [this](https://javadoc.io/static/io.smallrye.reactive/mutiny/1.0.0/io/smallrye/mutiny/groups/UniRepeat.html#until(java.util.function.Predicate)) for details about `until` and `whilst`.

Reply:
There's an error in the Javadoc BTW: `until` stops when the predicates passes (the Javadoc says the opposite).

Reply:
From reading the Javadoc, `whilst()` is really what we want.

Reply:
https://github.com/smallrye/smallrye-mutiny/pull/676